autoload/lsp/utils/workspace_edit.vim

```diff
@@ -1,12 +1,27 @@
 " Applies WorkspaceEdit changes.
 function! lsp#utils#workspace_edit#apply_workspace_edit(workspace_edit) abort
+    let l:loclist_items = []
+
     if has_key(a:workspace_edit, 'documentChanges')
         for l:text_document_edit in a:workspace_edit['documentChanges']
-            call lsp#utils#text_edit#apply_text_edits(l:text_document_edit['textDocument']['uri'], l:text_document_edit['edits'])
+            let l:loclist_items += s:_apply(l:text_document_edit['textDocument']['uri'], l:text_document_edit['edits'])
         endfor
     elseif has_key(a:workspace_edit, 'changes')
         for [l:uri, l:text_edits] in items(a:workspace_edit['changes'])
-            call lsp#utils#text_edit#apply_text_edits(l:uri, l:text_edits)
+            let l:loclist_items += s:_apply(l:uri, l:text_edits)
         endfor
     endif
+
+    if g:lsp_show_workspace_edits
+        call setloclist(0, l:loclist_items, 'r')
+        execute 'lopen'
+    endif
+endfunction
+
+"
+" _apply
+"
+function! s:_apply(uri, text_edits)
+    call lsp#utils#text_edit#apply_text_edits(a:uri, a:text_edits)
+    return lsp#utils#text_edit#build_loclist_items(a:uri, reverse(a:text_edits))
```

Comment:
I wonder if this `reverse()` is still needed. I remember that the original intention was to get the location list ordered by line number in ascending order. But I guess the order of workspace edit changes is entirely decided by the language server you happen to communicate with?

Would it make sense to do the sorting of text edits (grouped by file name, sorted by line number ascending) on the client?

Reply:
i don't have strong opinion for this now but if we do add `reverse` let us add why we do it for future reference.

Grouping would help but not required.

Reply:
I'll remove the reverse